### PR TITLE
ESlint changes

### PR DESCRIPTION
**Pull request by olivershooter**
This was supposed to be just a few changes to the eslint file (that picks up on small things and makes them an error, for example correct import order, turning used before defined, etc.) but turned into this mess of a refactor to:
a) make things work
b) make things work.

sorry davu

### CODE REVIEW

File: ui/src/main.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import { BrowserRouter } from 'react-router-dom';
-import App from './app/App';
+import Root from 'routes/root';
 import './index.css';
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
```
Comment (Dave-Turnbull) [inline ui/src/main.tsx:7]: why is this root?

File: ui/package.json [hunk 2]
```diff
@@ -22,41 +22,37 @@
...
-    "@vitest/coverage-v8": "^0.33.0",
-    "jsdom": "^22.1.0",
+    "@fontsource/roboto": "^5.0.8",
+    "@mui/icons-material": "^5.14.6",
+    "@mui/material": "^5.14.6",
+    "eslint-config-prettier": "^9.0.0",
+    "eslint-plugin-import": "^2.28.1",
+    "eslint-plugin-jsx-a11y": "^6.7.1",
+    "eslint-plugin-react": "^7.33.2",
+    "localforage": "^1.10.0",
```
Comment (olivershooter) [inline ui/package.json:32]: Get rid

File: ui/package.json [hunk 3]
```diff
@@ -22,41 +22,37 @@
...
+    "match-sorter": "^6.3.1",
```
Comment (olivershooter) [inline ui/package.json:33]: Get rid

File: ui/package.json [hunk 4]
```diff
@@ -22,41 +22,37 @@
...
+    "prettier": "^3.0.2",
```
Comment (olivershooter) [inline ui/package.json:34]: why?

File: ui/package.json [hunk 5]
```diff
@@ -22,41 +22,37 @@
...
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-router": "^6.14.1",
-    "react-router-dom": "^6.14.1",
-    "styled-components": "^5.3.11"
+    "react-router-dom": "^6.15.0",
+    "sort-by": "^1.2.0",
```
Comment (olivershooter) [inline ui/package.json:38]: Get rid

File: ui/package.json [hunk 6]
```diff
@@ -22,41 +22,37 @@
...
+    "@testing-library/react": "^14.0.0",
+    "@types/react": "^18.2.15",
+    "@types/react-dom": "^18.2.7",
+    "@typescript-eslint/eslint-plugin": "^6.4.1",
+    "@typescript-eslint/parser": "^6.4.1",
+    "@vitejs/plugin-react": "^4.0.3",
+    "eslint": "^8.47.0",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-react-refresh": "^0.4.3",
+    "jsdom": "^22.1.0",
```
Comment (olivershooter) [inline ui/package.json:52]: why

File: ui/package.json [hunk 7]
```diff
@@ -22,41 +22,37 @@
...
+    "pretty-quick": "^3.1.3",
```
Comment (olivershooter) [inline ui/package.json:53]: why

**Review by Dave-Turnbull — APPROVED** [review]

**Review by Dave-Turnbull — APPROVED** [review]

**Pull request merged**